src/mscorlib/src/System/String.Comparison.cs

```diff
@@ -25,42 +22,6 @@ public partial class String
         //Native Static Methods
         //
 
-        private unsafe static int CompareOrdinalIgnoreCaseHelper(String strA, String strB)
-        {
-            Debug.Assert(strA != null);
-            Debug.Assert(strB != null);
-            int length = Math.Min(strA.Length, strB.Length);
-
-            fixed (char* ap = &strA._firstChar) fixed (char* bp = &strB._firstChar)
-            {
-                char* a = ap;
-                char* b = bp;
-                int charA = 0, charB = 0;
-
-                while (length != 0)
-                {
-                    charA = *a;
-                    charB = *b;
-
-                    Debug.Assert((charA | charB) <= 0x7F, "strings have to be ASCII");
```

Comment:
We are now replacing calls to this method (CompareOrdinalIgnoreCaseHelper) with calls to `CompareOrdinalIgnoreCase(ReadOnlySpan<char> strA, ReadOnlySpan<char> strB)`

Should this be 0x7F instead of 0x80?

https://github.com/dotnet/coreclr/blob/master/src/mscorlib/shared/System/Globalization/CompareInfo.cs#L569

```C#
// in InvariantMode we support all range and not only the ascii characters.
char maxChar = (char) (GlobalizationMode.Invariant ? 0xFFFF : 0x80);
```

Also, this method doesn't do the `(*a <= maxChar) && (*b <= maxChar)` checks for each character. Wouldn't we see perf regression now (including the overhead of casting string to span)?

Reply:
I think the check needs to be `<= 0x7F` and not `<= 0x80` to remain consistent with the IsAscii checks.

https://github.com/dotnet/coreclr/blob/ba1d5b2de899bf6e09702f70c28e85b527bd1ea9/src/vm/object.cpp#L1392

cc @tarekgh, @jkotas

Reply:
Yeah, that;s not write; either should be `< maxChar` or `maxChar` should be `0x7f`; since the alternative is `0xffff` guessing should be `0x7f`

Reply:
https://github.com/dotnet/coreclr/pull/16535